src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,304 @@
+            <div main-title>${this.hass.localize("panel.calendar")}</div>
+          </app-toolbar>
+        </app-header>
+        <div class="content">
+          <div class="calendar-list">
+            <ha-card header="Calendars">
+              ${this._calendars.map(
+                (selCal) =>
+                  html`<div class="calendar-toggle">
+                    <paper-checkbox
+                      style=${styleMap({
+                        "--primary-color": selCal.backgroundColor,
+                      })}
+                      .value=${selCal.calendar.entity_id}
+                      .checked=${selCal.selected}
+                      @change=${this._handleToggle}
+                    >
+                      ${selCal.calendar.name}</paper-checkbox
+                    >
+                  </div>`
+              )}
+            </ha-card>
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private _fetchCalendars() {
+    this.hass.callApi<Calendar[]>("GET", "calendars").then((result) => {
+      this._calendars = result.map((cal, idx) => ({
+        selected: true,
+        calendar: cal,
+        backgroundColor: `#${palette[idx]}`,
+      }));
+    });
+  }
+
+  private _fetchData() {
+    if (!this._start || !this._end) {
+      return;
+    }
+
+    if (!this._firstUpdate && this._calendars.length) {
+      this._firstUpdate = true;
+    }
+
+    const start = new Date(this._start);
+    const end = new Date(this._end);
+    const params = encodeURI(
+      `?start=${start.toISOString()}&end=${end.toISOString()}`
+    );
+
+    const calEvents: CalendarEvent[] = [];
+    const calls = new Promise((resolve) => {
+      this._calendars
+        .filter((selCal) => selCal.selected)
+        .forEach(async (selCal, idx) => {
+          const events = await this.hass.callApi<any[]>(
+            "GET",
+            `calendars/${selCal.calendar.entity_id}${params}`
+          );
+          events.forEach((ev) => {
+            const eventStart = this._getDate(ev.start);
+            if (!eventStart) {
+              return;
+            }
+            const eventEnd = this._getDate(ev.end);
+            const event: CalendarEvent = {
+              start: eventStart,
+              end: eventEnd,
+              title: ev.summary,
+              summary: ev.summary,
+              backgroundColor: selCal.backgroundColor,
+              borderColor: selCal.backgroundColor,
+            };
+
+            calEvents.push(event);
+
+            if (idx === this._calendars.filter((c) => c.selected).length - 1) {
+              resolve();
+            }
+          });
+        });
+    });
+
+    calls.then(() => {
+      this._events = calEvents;
+    });
+  }
+
+  private _getDate(dateObj: any): string | undefined {
+    if (typeof dateObj === "string") {
```

Comment:
This is also kind of a bandaid. Need to update the demo calendar to return events in the right format

Reply:
I think this is a call to the mock API. Not sure how to resolve

Reply:
You should change it in the backend demo, but you would have to check all the calendar integrations.